crates/README.md

```diff
@@ -0,0 +1,147 @@
+- `roc_mono` Roc's main intermediate representation (IR), which is responsible for monomorphization, defunctionalization, inserting ref-count instructions, and transforming a Roc program into a form that is easy to consume by a backend.
+- `roc_parse` Implements the Roc parser, which transforms a textual representation of a Roc program to an abstract syntax tree.
+- `roc_problem` provides types to describe problems that can occur when compiling `.roc` code.
+- `roc_region` Data structures for storing source-code-location information, used heavily for contextual error messages.
+- `roc_target` provides types and helpers for compiler targets such as `default_x86_64`.
+- `roc_serialize` provides helpers for serializing and deserializing to/from bytes.
+- `roc_solve` The entry point of Roc's type inference system. Implements type inference and specialization of abilities.
+- `roc_solve_problem` provides types to describe problems that can occur during solving.
+- `roc_str` provides `Roc` styled collection reference counting. See [README.md](./compiler/str/README.md) for more information.
+- `test_derive` Tests Roc's auto-derivers.
+- `test_gen` contains all of Roc's code generation tests. See [README.md](./compiler/test_gen/README.md) for more information.
+- `test_mono` Tests Roc's generation of the mono intermediate representation.
+- `test_mono_macros` Macros for use in test_mono
+- `roc_types` Various representations and utilities for dealing with types in the Roc compiler.
+- `roc_unify` Implements Roc's unification algorithm, the heartstone of Roc's type inference.
+
+## `docs/` - `roc_docs`
+
+Generates html documentation from Roc files.
+Used for [roc-lang.org/builtins/Num](https://www.roc-lang.org/builtins/Num).
+
+## `docs_cli/` - `roc_docs_cli` library and `roc-docs` binary
+
+Provides a binary that is only used for static build servers.
+
+## `editor/` - `roc_editor`
+
+For editing Roc files (Work In Progress). See [README.md](./editor/README.md) for more information.
+
+## `error_macros/` - `roc_error_macros`
+
+Provides macros for consistent reporting of errors in Roc's rust code.
+
+## `glue/` - `roc_glue`
+
+The `roc_glue` crate generates rust code to connect a Roc app with a rust platform. This tool is not necessary for writing a platform in another language, however, it's an added convenience for rust platform devs.
+
+## `highlight/` - `roc_highlight`
+
+Provides syntax highlighting for the editor by transforming a string to markup nodes.
+
+## `linker/` - `roc_linker`
+
+Surgical linker that links platforms to Roc applications.
+We created our own linker for speed and because regular linkers add a lot of problems and complexity. Because we want roc to manage the build system and final linking of the executable, it is significantly less practical to use a regular linker. This can be seen in how many arbitrary libraries we link with the legacy linker because we have no idea what libraries the application is actually using.
```

Comment:
```suggestion
We created our own linker for performance, since regular linkers add complexity that is not needed for linking Roc apps. Because we want `roc` to manage the build system and final linking of the executable, it is significantly less practical to use a regular linker.
```